src/v2/Apps/Search/Components/NavigationTabs.tsx

```diff
@@ -59,88 +106,67 @@ export class NavigationTabs extends React.Component<Props> {
     const tabName = text.replace(/[0-9]/g, "").trim()
 
     return (
-      <RouteTab
+      <RoundedRouteTab
         to={to}
+        text={tabName}
+        key={tabName}
         exact={exact}
+        count={count}
         onClick={event => {
           event.preventDefault()
-          // @ts-expect-error STRICT_NULL_CHECK
-          this.props.router.push(to)
-          this.trackClick(tabName, to)
+          router.router.push(to)
+          trackClick(tabName, to)
         }}
-        key={to}
-      >
-        {text}
-        {count != null && (
-          <Text variant="text" display="inline">
-            &nbsp;({count})
-          </Text>
-        )}
-      </RouteTab>
+      />
     )
   }
 
-  tabs() {
-    const { term, artworkCount } = this.props
-
-    const route = tab => `/search${tab.replace(/\s/g, "_")}?term=${term}`
-
-    let restAggregationCount: number = 0
-    MORE_TABS.forEach(
-      key =>
-        // @ts-expect-error STRICT_NULL_CHECK
-        (restAggregationCount += get(
-          aggregationFor(this.props, key),
-          // @ts-expect-error STRICT_NULL_CHECK
-          agg => agg.count,
-          0
-        ))
-    )
+  const tabs: JSX.Element[] = []
 
-    const tabs = []
+  artworkCount > 0 &&
+    tabs.push(
+      renderTab("Artworks", route(""), {
+        count: artworkCount,
+        exact: true,
+      })
+    )
 
-    !!artworkCount &&
+  Object.entries(tabCountMap(searchableConnection)).map(
+    ([key, value]: [string, number]) => {
       tabs.push(
-        // @ts-expect-error STRICT_NULL_CHECK
-        this.renderTab("Artworks", route(""), {
-          count: artworkCount,
-          exact: true,
+        renderTab(key, route(`/${key.toLowerCase()}`), {
+          count: value,
         })
       )
+    }
+  )
 
-    Object.entries(tabCountMap(this.props)).map(
-      ([key, value]: [string, number]) => {
-        tabs.push(
-          // @ts-expect-error STRICT_NULL_CHECK
-          this.renderTab(key, route(`/${key.toLowerCase()}`), {
-            count: value,
-          })
-        )
-      }
+  const restAggregationCount = MORE_TABS.reduce((prev, key) => {
+    const tabAggregation = get(
+      aggregationFor(searchableConnection, key),
+      agg => agg?.count,
+      0
     )
 
-    !!restAggregationCount &&
-      tabs.push(
-        // @ts-expect-error STRICT_NULL_CHECK
-        this.renderTab("More", route("/more"), {
-          count: restAggregationCount,
-        })
-      )
-
-    return tabs
-  }
+    return tabAggregation ? (prev += tabAggregation) : prev
+  }, 0)
 
-  render() {
-    return (
-      <RouteTabs key={`tab-carousel-${this.props.term}`}>
-        {this.tabs()}
-      </RouteTabs>
+  restAggregationCount > 0 &&
+    tabs.push(
+      renderTab("More", route("/more"), {
+        count: restAggregationCount,
+      })
     )
-  }
+
+  return (
+    <Swiper my={2} snap="start">
```

Comment:
I'm sure this is fine; though generally we use this for things of a consistent width. It might be more appropriate to just use `BaseTabs` here — which is a free scroll with no snapping but has the gradient fade overlay as an affordance when items go off-screen. I _think_ it should work fine. I may be mistaken.

Reply:
Thanks Damon, but when I use BaseTabs I get a separator at the bottom of tabs
![image](https://user-images.githubusercontent.com/56556580/133278346-8097d88a-4ea8-457e-8b8e-05df6dd6492a.png)

And I don't understand how I can remove it. Can you help me with it?

Reply:
Ah, OK maybe punt on this for now and I'll extract the underlying primitive so that we can follow up. Thanks for looking!